paper.md

```diff
@@ -0,0 +1,400 @@
+   index: 5
+ - name: Center for the Study of Complex Systems, University of Michigan
+   index: 6
+ - name: Computer Science, Grand Valley State University
+   index: 7
+date: 30 July 2021
+bibliography: paper.bib
+breaks: false
+---
+
+<!-- [Scraps document here](https://hackmd.io/@DSAbJHksRtqYhjQ4NWzmGA/S1z3oQDWv) -->
+<!--  @AML (2021-06-12): Doing a comment clean up, moving important ones (e.g., old meeting notes) into the scraps document. -->
+
+<!-- INSTRUCTIONS
+
+- The paper should be between 250-1000 words.
+
+Paper should include:
+
+- A summary describing the high-level functionality and purpose of the software for a diverse, non-specialist audience.
+- A clear Statement of Need that illustrates the research purpose of the software.
+- A list of key references, including to other software addressing related needs.
+- Mention (if applicable) a representative set of past or ongoing research projects using the software and recent scholarly publications enabled by it.
+- Acknowledgement of any financial support.
+
+-->
+
+<!-- RESOURCES
+
+- Empirical coding guidelines: https://github.com/devosoft/Empirical/wiki/Coding-Guidelines
+- Empirical README: https://github.com/devosoft/Empirical/blob/master/README.md
+
+-->
+
+# Summary
+<!-- SUMMARY
+
+JOSS: "Begin your paper with a summary of the high-level functionality of your software for a non-specialist reader. Avoid jargon in this section."
+
+i.e., What does this software do?
+-->
+
+# Summary
+
+Empirical is a C++ library designed to promote open science and facilitate the development of scientific software that is efficient, reliable, and easily distributable to researchers and non-experts alike.
+Specifically, the library sets out to fulfill the following goals:
+
+1. **Utility:** Empirical features a wide selection of helper tools to streamline common scientific computing tasks such as configuration, data management, random number generation, and mathematical manipulations. Documentation, examples, and project templates streamline developer onboarding.
+2. **Efficiency:** Empirical tools emphasize efficiency to make heavy scientific computing workloads tractable. Where possible, computation is moved to compile-time, heap allocation is avoided, and memory-locality is respected.
+3. **Reliability:** Scientific simulation and data analysis is meaningful insofar as it is correct. In conjunction with extensive unit testing, Empirical benefits from --- and provides --- sophisticated debug-mode instrumentation tools. These include audited memory management, drop-in replacements for standard C++ containers with safety checking, and self-documenting assertions.
+4. **Distributability:** Scientific software should be easily used by any researcher, student, or citizen scientist who wants to experiment, explore, or replicate studies. As such, Empirical is highly portable, uses common data formats, facilitates flexible runtime configuration, and simplifies compilation to a performant web app.
+
```

Comment:
```suggestion

The core Empirical support library comprises four major components:

 1. debug-instrumented fundamental type and STL container wrappers,
 2. implementations of general-purpose data structures and algorithms,
 3. integrated, end-to-end frameworks for data and configuration management, and
 4. object-oriented bindings for Emscripten/WebAssembly GUI elements.
```

Reply:
if first mention of STL:
"STL" ==> standard template library (STL)

Reply:
per cpp reference, it looks the official name is actually “standard library”! will change and not have acronym

Reply:
Somehow I completely missed that the STL was never official and that the C++ Standard Library was heavily based off of it, but is technically something new with C++ 11.